src/Uno.Extensions.Configuration/ConfigurationBinder.cs

```diff
@@ -0,0 +1,731 @@
+					}
+				}
+			}
+		}
+
+		return instance;
+	}
+
+	private static object CreateInstance(
+
+		Type type)
+	{
+		if (type.IsInterface || type.IsAbstract)
+		{
+			throw new InvalidOperationException("error");
+		}
+
+		if (type.IsArray)
+		{
+			if (type.GetArrayRank() > 1)
+			{
+				throw new InvalidOperationException("error");
+			}
+
+			return Array.CreateInstance(type.GetElementType(), 0);
+		}
+
+		if (!type.IsValueType)
+		{
+			bool hasDefaultConstructor = type.GetConstructors(DeclaredOnlyLookup).Any(ctor => ctor.IsPublic && ctor.GetParameters().Length == 0);
+			if (!hasDefaultConstructor)
+			{
+				throw new InvalidOperationException("error");
+			}
+		}
+
+		try
+		{
+			return Activator.CreateInstance(type);
+		}
+		catch (Exception ex)
+		{
+			throw new InvalidOperationException("error", ex);
+		}
+	}
+
+	private static void BindDictionary(
+		object dictionary,
+			Type dictionaryType,
+		IConfiguration config, BinderOptions options)
+	{
+		// IDictionary<K,V> is guaranteed to have exactly two parameters
+		Type keyType = dictionaryType.GenericTypeArguments[0];
+		Type valueType = dictionaryType.GenericTypeArguments[1];
+		bool keyTypeIsEnum = keyType.IsEnum;
+
+		if (keyType != typeof(string) && !keyTypeIsEnum)
+		{
+			// We only support string and enum keys
+			return;
+		}
+
+		PropertyInfo setter = dictionaryType.GetProperty("Item", DeclaredOnlyLookup);
+		foreach (IConfigurationSection child in config.GetChildren())
+		{
+			object item = BindInstance(
+				type: valueType,
+				instance: null!,
+				config: child,
+				options: options);
+			if (item != null)
+			{
+				if (keyType == typeof(string))
+				{
+					string key = child.Key;
+					setter.SetValue(dictionary, item, new object[] { key });
+				}
+				else if (keyTypeIsEnum)
+				{
+					object key = Enum.Parse(keyType, child.Key);
+					setter.SetValue(dictionary, item, new object[] { key });
+				}
+			}
+		}
+	}
+
+	private static void BindCollection(
+		object collection,
+			Type collectionType,
+		IConfiguration config, BinderOptions options)
+	{
+		// ICollection<T> is guaranteed to have exactly one parameter
+		Type itemType = collectionType.GenericTypeArguments[0];
+		MethodInfo addMethod = collectionType.GetMethod("Add", DeclaredOnlyLookup);
+
+		foreach (IConfigurationSection section in config.GetChildren())
+		{
+			try
+			{
+				object item = BindInstance(
+					type: itemType,
+					instance: null!,
+					config: section,
+					options: options);
+				if (item != null)
+				{
+					addMethod.Invoke(collection, new[] { item });
+				}
+			}
+			catch
+			{
+			}
+		}
+	}
+
+	private static object BindImmutableList(
+		object collection,
+			Type collectionType,
+		IConfiguration config, BinderOptions options)
+	{
+		// ICollection<T> is guaranteed to have exactly one parameter
+		Type itemType = collectionType.GenericTypeArguments[0];
+		MethodInfo addMethod = collectionType.GetMethod("Add", DeclaredOnlyLookup);
```

Comment:
Should be great to use the `Builder` when available (NTH - the real fix would be to use the `System.Text.Json`).